core/src/test/scala/unit/kafka/server/ServerStartupTest.scala

```diff
@@ -62,7 +65,7 @@ class ServerStartupTest extends ZooKeeperTestHarness {
       TestUtils.createServer(KafkaConfig.fromProps(props2))
       fail("Starting a broker with the same port should fail")
     } catch {
-      case _: RuntimeException => // expected
+      case _: BindException | NodeExistsException => // expected
```

Comment:
BindException is the cause not the exception thrown. What's thrown is a KafkaException. So, you will want to catch that. Also, we can get rid of NodeExistsException.

Reply:
Ah thanks for catching it, changing to KafkaException